### PR TITLE
*: solve cloud tikv disk full problem

### LINKED ISSUE #243: Solve cloud TiKV disk full problem - Client Go part

### PR DESCRIPTION
**Pull request by tier-cap**
### What problem does this PR solve?

Issue Number：closes #243

When the disk is full, no matter one or two or the majority or all, what we expected will be:

1. TiKV not crash directly.
2. Support restoring storage through drop table/delete from table, expansion is also allowed.
3. Provide slightly better reading ability.

###  What is changed and how it works?

1. Introduce a two-level threshold defense mechanism
2. Different flow control algorithms are used under each threshold to provide different business support capabilities and spatial recovery capabilities

### Check List

Test

UserDoc

### Release note

- No release note.

### CODE REVIEW

File: tikv/txn.go [hunk 1]
```diff
@@ -307,6 +309,10 @@ func (txn *KVTxn) SetKVFilter(filter KVFilter) {
 	txn.kvFilter = filter
 }
 
+func (txn *KVTxn) SetAllowed() {
```
Comment (sticnarf) [inline tikv/txn.go:312]: ditto

File: tikv/2pc.go [hunk 2]
```diff
@@ -755,6 +759,10 @@ func (c *twoPhaseCommitter) keySize(key, value []byte) int {
 	return len(key)
 }
 
+func (c *twoPhaseCommitter) SetAllowed() {
```
Comment (sticnarf) [inline tikv/2pc.go:762]: Please use the full name `SetAllowedOnDiskFull`

File: tikv/txn.go (hunk 1 above)
Comment (sticnarf) [inline tikv/txn.go:312]: ditto

File: go.mod [hunk 3]
```diff
@@ -44,3 +44,4 @@ require (
 	gopkg.in/yaml.v2 v2.4.0 // indirect
 	honnef.co/go/tools v0.2.0 // indirect
 )
+replace github.com/pingcap/kvproto => /home/xiaojin.zxj/010.tidb/kvproto
```
Comment (iosmanthus) [inline go.mod:47]: Please use the repo of your own instead of the local path. The other couldn't test it with the local path.

**Comment by disksing** [comment]
Looks no progress for a long time. @tier-cap I'll close it for now to avoid distraction. You can reopen the PR when it is ready.

**Pull request closed without merging**